vm.go

```diff
@@ -306,12 +306,32 @@ type vm struct {
 	interrupted   uint32
 	interruptVal  interface{}
 	interruptLock sync.Mutex
+
+	lastFunctionTicks uint64
 }
 
 type instruction interface {
 	exec(*vm)
 }
 
+func (vm *vm) setPrg(prg *Program) {
+	if vm.r.functionTickTrackingEnabled {
+		currentTicks := vm.r.Ticks()
+		if vm.prg != nil {
+			if vm.r.functionTickMetrics == nil {
+				vm.r.functionTickMetrics = make(map[string]uint64)
+			}
```

Comment:
[nit] Can we initialize this in the `(r *Runtime) init()` function instead to avoid performing this check for each execution? We can probably initialize it only when `functionTickTrackingEnabled` is true.